application/src/main/java/org/opentripplanner/street/model/edge/StreetEdge.java

```diff
@@ -1104,9 +1113,23 @@ private TraversalCosts otherTraversalCosts(
   private TraversalCosts bicycleOrScooterTraversalCost(
     StreetSearchRequest req,
     TraverseMode mode,
-    double speed
+    double speed,
+    State state
   ) {
-    double time = getEffectiveBikeDistance() / speed;
+    PropulsionType propulsion = state.isRentingVehicle()
+      ? state.rentalVehiclePropulsionType()
+      : null;
+
+    double electricAssistSlopeSensitivity = mode == TraverseMode.BICYCLE
+      ? req.bike().rental().electricAssistSlopeSensitivity()
+      : req.scooter().rental().electricAssistSlopeSensitivity();
```

Comment:
This can be a method on `StreetSearchRequest`.

Reply:
I think I did what you wanted now.